.babelrc

```diff
@@ -1,4 +1,11 @@
 {
-  "presets": ["stage-0", "es2015", "react"],
-  "plugins": ["transform-decorators-legacy"]
+  "presets": [
+    "@babel/preset-env",
+    "@babel/preset-react"
+  ],
+  "plugins": [
+    "@babel/plugin-syntax-dynamic-import",
+    "@babel/plugin-proposal-class-properties",
```

Comment:
I'm not quite familiar with the Babel stack now, can you explain why these two plugins are required?

Reply:
Both are official Babel plugins. Since Babel 7, Babel is trying to move to a [scope system](https://docs.npmjs.com/misc/scope) that's why I have to prefix `@babel` to the plugin names.

**@babel/plugin-syntax-dynamic-import**: is used for situations for using dynamic imports `import()` syntax; I guess I forgot to remove this as we're not doing any.

**@babel/plugin-proposal-class-properties**: is for new [class fields](https://github.com/tc39/proposal-class-fields) where we can define any property on the class as a regular assignment. This one is required, because, instead of using `::this.handleSomeChange` to bind `this`, I'm using the class fields and arrow functions to do so. The syntax is cleaner and more readable. This may look personal preference but class fields syntax is at stage-3 and more likely to land to JavaScript world whereas `::` binding which is at stage-0.

We can write this:
```js
class MyClass {
  ...
  handleSomeChange() {
    doSomething(this.state.value);
  }
  ...
  render() {
    return (
      <button onClick={::this.handleSomeChange}>Click</button>
    )
  }
}
```

To this:
```js
class MyClass {
  ...
  handleSomeChange = () => {
    doSomething(this.state.value);
  }
  ...
  render() {
    return (
      <button onClick={this.handleSomeChange}>Click</button>
    )
  }
}
```

I guess that explains why I have added this plugin.
I'm removing `@babel/plugin-syntax-dynamic-import` because we're not using it anyway; Thanks for pointing out.

Reply:
thanks for your excellent explanation!